### PR TITLE
Create yaml template for openstack-mkcloud job.

### PR DESCRIPTION
**Pull request by gosipyan**
I have created temporary job in Jenkins https://ci.suse.de/job/gosipyan-openstack-mkcloud/.

### CODE REVIEW

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml [hunk 1]
```diff
@@ -0,0 +1,489 @@
...
+               - run-update
+               - scm-tag
+          anonymous:
+               - job-read
+
+    parameters:
+     - string:
+        name: cloudsource
+        default: develcloud5
+        description: "Can be something like develcloud3 (uses Devel:Cloud:*, see TESTHEAD parameter for Staging),  susecloud3 (uses SUSE:*), GM3 (uses the GM ISO)"
```
Comment (aplanas) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: Not related (and unimportant), but maybe you can update the comment to refer cloud6 or cloud7 here

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml [hunk 2]
```diff
@@ -0,0 +1,489 @@
...
+               - run-delete
+               - run-update
+               - scm-tag
+          anonymous:
+               - job-read
+
+    parameters:
+     - string:
+        name: cloudsource
+        default: develcloud5
```
Comment (aplanas) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: move the default to `develcloud7`??

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml [hunk 3]
```diff
@@ -0,0 +1,489 @@
...
+        description: "Can be something like develcloud3 (uses Devel:Cloud:*, see TESTHEAD parameter for Staging),  susecloud3 (uses SUSE:*), GM3 (uses the GM ISO)"
+
+     - string:
+        name: mkcloudtarget
+        default: all
+        description:  Set mkcloudtarget to define the mkcloud command line parameters .
```
Comment (aplanas) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: there are two extra spaces: one at the beginning and other before the dot.

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml [hunk 4]
```diff
@@ -0,0 +1,489 @@
...
+     - choice:
```
Comment (aplanas) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: Can you separate the parameter with an empty line?

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml [hunk 5]
```diff
@@ -0,0 +1,489 @@
...
+         name: libvirt_typ
```
Comment (aplanas) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: `libvirt_type`

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml [hunk 6]
```diff
@@ -0,0 +1,489 @@
...
+            - openvswitch
+            - linuxbridge
+
+     - choice:
+         name: networkingmode
+         choices:
+             - gre
+             - vlan
+             - vxlan
+     - choice:
```
Comment (aplanas) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: separation

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml [hunk 7]
```diff
@@ -0,0 +1,489 @@
...
+         name: storage_method
+         description: 'Select storage method: default (let mkcloud decide), none (enforce none)'
```
Comment (aplanas) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: are quotes needed? if so you need to update the rest of the descriptions

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml [hunk 8]
```diff
@@ -0,0 +1,489 @@
...
+     
+     - string:
+         name: TESTHEAD
+         default : '1'
+         description: 'if non-empty, test latest version from Devel:Cloud (:1.0)'
+   
+     - string:
+         name: nodenumber
+         default: '2'
+         description: number of nodes excl. admin-node minimum 2=controller+compute
```
Comment (aplanas) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: quotes? (there are more)

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml [hunk 9]
```diff
@@ -0,0 +1,489 @@
...
+
+     - string:
+         name: nodenumbercompute
+         default:
+         description: Number of compute  nodes
```
Comment (aplanas) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: extra space between compute and nodes

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml [hunk 10]
```diff
@@ -0,0 +1,489 @@
...
+
+     - string: 
+         name: want_tempes
```
Comment (aplanas) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: `want_tempest`

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml [hunk 11]
```diff
@@ -0,0 +1,489 @@
...
+         default: 
+         description: can be used for force off(0) or on(1)
+
+     - string: 
+         name: tempestoptions
+         default:
```
Comment (aplanas) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: missing default values: `-t -s`

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml [hunk 12]
```diff
@@ -0,0 +1,489 @@
...
+         description: |
+           Additional options to pass to the \"run_tempest.sh\
+           \ -N \" step.\nUse just \"-t\" for a full tempest run.\nDefault: -t -s.
+
+     - string: 
+         name: ostestroptions
+         default: -t -s
```
Comment (aplanas) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: this doesn't have default values

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml [hunk 13]
```diff
@@ -0,0 +1,489 @@
...
+
+     - string: 
+         name: want_all_ssl
+         default:
+         description: set to 1 to deploy all cloud services with SSL aka HTTPS
+
+     - choice: 
+         name: cinder_backend
+         choices:
+           - local
```
Comment (aplanas) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: the original one have also the empty line as a valid choice, nos sure if needs to be added here

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml [hunk 14]
```diff
@@ -0,0 +1,489 @@
...
+         default: 
+         description: Set to UEFI boot admin node
+        
+     - string:
+         name: want_murano
+         default: 
+         description: set to 1 to deploy murano
+
+     - string:
+         name: want_horizon_integration_test
```
Comment (aplanas) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: duplicated?

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml [hunk 15]
```diff
@@ -0,0 +1,489 @@
...
+     - string:
+         name: want_efi
+         default: 
+         description: Set to UEFI boot admin node
+        
+     - string:
+         name: want_murano
+         default: 
+         description: set to 1 to deploy murano
+
```
Comment (aplanas) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: Around here there is a missing `custom_settings` multi-line parameter in the original

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml [hunk 16]
```diff
@@ -0,0 +1,489 @@
...
+     - build-name:
+         name: '#${BUILD_NUMBER}: ${ENV,var="cloudsource"} (${ENV,var="nodenumber"}/${ENV,var="networkingplugin"}/${ENV,var="tempestoptions"}/${ENV,var="mkcloudtarget"}/${ENV,var="github_pr"}${ENV,var="hacloud"})'
+     - timeout:
+        timeout: 3600
+        timeout-var: 'BUILD_TIMEOUT'
+        fail: true
+        type: absolute
+
+    builders:
+    - shell: |+
```
Comment (aplanas) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: Can you check if this script is OK after calling JJB? IIRC we need to scape certain things

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml [hunk 17]
```diff
@@ -0,0 +1,489 @@
...
+            exit $ret # check return code before tee
+        fi
+
+        # report mkcloud-gating status
+        if [[ $github_pr_sha ]] ; then
+          $ghs -r $github_pr_repo -a set-status -s "success" -t $BUILD_URL -c $github_pr_sha --context $ghs_context
+          trap "-" ERR
+        fi
+
+
```
Comment (aplanas) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: too many extra lines : )

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml [hunk 18]
```diff
@@ -0,0 +1,489 @@
...
+        if [[ $github_pr_sha ]] ; then
+          $ghs -r $github_pr_repo -a set-status -s "success" -t $BUILD_URL -c $github_pr_sha --context $ghs_context
+          trap "-" ERR
+        fi
+
+       
+    publishers:
+      - archive:
+         artifacts: .artifacts/**
+         allow-empty: 'false'
```
Comment (aplanas) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: not sure that this needs to be a string, but directly a boolean

**Comment by aplanas** [comment]
@gosipyan thanks for the conversion, and we are very close now : )

**Comment by aplanas** [comment]
Maybe we also need to remove openstack-mkcloud.xml after this is merged? so we avoid confusions of what file to maintain.

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml [hunk 19]
```diff
@@ -0,0 +1,489 @@
+- job:
+    name: openstack-mkcloud
+    node: openstack-mkcloud
+    description: |
+     "!!! This job is automatically pushed from GIT to ci.suse.de !!!
```
Comment (gosipyan) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: Fixed

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml [hunk 20]
```diff
@@ -0,0 +1,489 @@
...
+    properties:
+      - authorization:
+          cloud:
+               - job-build
+               - job-cancel
+               - job-configure
+               - job-delete
+               - job-discover
+               - job-read
+               - job-status
```
Comment (gosipyan) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: its in https://ci.suse.de/job/openstack-mkcloud/configure Enable project-based security section.

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml (hunk 2 above)
Comment (gosipyan) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: Fixed

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml (hunk 1 above)
Comment (gosipyan) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: Fixed

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml (hunk 3 above)
Comment (gosipyan) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: Fixed

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml (hunk 6 above)
Comment (gosipyan) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: Fixed

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml (hunk 7 above)
Comment (gosipyan) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: I change to | and text move to next line.Will update all.Without quotes it fails.

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml (hunk 8 above)
Comment (gosipyan) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: I get following error without quotes . TypeError: cannot serialize 1 (type int) when run jjb test.

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml (hunk 9 above)
Comment (gosipyan) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: Fixed

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml (hunk 10 above)
Comment (gosipyan) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: Fixed

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml (hunk 11 above)
Comment (gosipyan) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: Fixed

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml (hunk 12 above)
Comment (gosipyan) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: Fixed

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml (hunk 13 above)
Comment (gosipyan) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: Fixed

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml (hunk 15 above)
Comment (gosipyan) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: Fixed

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml (hunk 17 above)
Comment (gosipyan) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: Fixed

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml (hunk 20 above)
Comment (gosipyan) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: Sorry my mistake.Fixed.

**Comment by rsalevsky** [comment]
Pls merge this after the milestone.

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml [hunk 21]
```diff
@@ -0,0 +1,508 @@
...
+     - build-name:
+         name: '#${BUILD_NUMBER}: ${ENV,var="cloudsource"} (${ENV,var="nodenumber"}/${ENV,var="networkingplugin"}/${ENV,var="tempestoptions"}/${ENV,var="mkcloudtarget"}/${ENV,var="github_pr"}${ENV,var="hacloud"})'
+     - timeout:
+        timeout: 3600
+        timeout-var: 'BUILD_TIMEOUT'
+        fail: true
+        type: absolute
+
+    builders:
+    - shell: |+
```
Comment (dirkmueller) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: extra + at the end of line

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml [hunk 22]
```diff
@@ -0,0 +1,508 @@
...
+      REMEMBER: changes to this job will be overwritten after a few hours - make sure to create a PR for your changes
+      This job runs https://github.com/SUSE-Cloud/automation/blob/master/scripts/mkcloud results are 
+      analyzed by https://github.com/SUSE-Cloud/automation/blob/master/scripts/jenkins/mkcloudresultcheck.pl
+      on mkcloudresultcheck@mkcloudresultcheck.cloud.suse.de every 15 minutes"
+
+    logrotate:
+        numToKeep: 1000
+        daysToKeep: 300
+
+    properties:
```
Comment (toabctl) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: Is that really needed?

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml [hunk 23]
```diff
@@ -0,0 +1,508 @@
...
+         default: 
+         description: can be used for force off(0) or on(1)
+
+     - string: 
+         name: tempestoptions
+         default: -t -s
+         description: |
+           Additional options to pass to the "run_tempest.sh -N " step.
+           Use just \"-t\" for a full tempest run.
+           Default: -t -s.
```
Comment (toabctl) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: nit: no need to write the defaults in the description I think. It's written 4 lines above

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml [hunk 24]
```diff
@@ -0,0 +1,508 @@
...
+         description: set to 1 to deploy murano
+
+     - text:
+         name: custom_settings
+         default: 
+         description: |
+              This holds shell code and will be executed before calling mkcloud. This should not be used by default or regularly. It is meant to test PRs with special configurations.
+
+     - string:
+         name: want_horizon_integration_test
```
Comment (toabctl) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: It's already there in line 127

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml (hunk 23 above)
Comment (gosipyan) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: Fixed

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml (hunk 24 above)
Comment (gosipyan) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: Fixed

File: scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml (hunk 22 above)
Comment (gosipyan) [inline scripts/jenkins/jobs-ibs/openstack-mkcloud.yaml]: @toabctl  when i remove   properties:   JJB test fails.

**Comment by gosipyan** [comment]
Updated directory to jenkins/ci.suse.de/openstack-mkcloud.yaml.

**Comment by jdsn** [comment]
@gosipyan @dirkmueller 
concurrent builds = false  is not a good idea!!

**Pull request merged**